required/tools/array.dtx

```diff
@@ -3285,12 +3298,23 @@ Bug reports can be opened (category \texttt{#1}) at\\%
 %    simple length expression, but with the calc package involved it
 %    could break under an "\edef" operation, which is how the preamble
 %    is constructed. We now make use of "\unexpanded" here to prevent that. The
-%    "\expandafter" gymnastics is necessary to expand the "#1" at least
+%    "\expandafter" gymnastics is necessary to expand the "#1" exactly
 %    once (since it will get "\@nextchar" as its value and need its
-%    content!
+%    content). However, once added to the preamble, no further
+%    expansion should happen if the preamble is extended. Therefore we
+%    change from \cs{@startpbox} to \cs{@startpbox@action}. The latter
+%    then has a trivial definition and expands (including its
+%    argument) to itself while the preamble is being built. Outside of this
+%    preamble built-up \cs{@startpbox@action} that does the actual work.
+%    The definition for \cs{@startpbox} is only ever used  during
+% \changes{v2.6h}{2024/12/17}{Expand argument of \cs{@startpbox}
+%      exactly once while the preamble is built and use
+%      \cs{@startpbox@action} later to do the real work (gh/1585)}
```

Comment:
Something strange happened here (and in the line above), looks like you rewrote that sentence and forgot to delete the old version.